src/main/java/com/faforever/client/io/ChecksumMismatchException.java

```diff
@@ -0,0 +1,22 @@
+package com.faforever.client.io;
+
+import lombok.Getter;
+
+public class ChecksumMismatchException extends Exception {
+  @Getter
+  private final Object item;
+  @Getter
+  private final String expected;
+  @Getter
+  private final String actual;
+
+  public ChecksumMismatchException(Object item, String expected, String actual) {
+    this.item = item;
+    this.expected = expected;
+    this.actual = actual;
+  }
```

Comment:
```suggestion
@Getter
@RequiredArgsConstructor
public class ChecksumMismatchException extends Exception {
  private final Object item;
  private final String expected;
  private final String actual;
```

Reply:
Is object really the most restrictive type? Couldn't it be something like File or Path?